src/modules/Experiences/ExperiencesList/helpers.js

```diff
@@ -84,4 +84,57 @@ const configureForm = (expType) => {
   }
 };
 
-export { formatMonthYearDate, getWorkLength, configureForm };
+const monthScore = {
+  January: 1,
+  February: 2,
+  March: 3,
+  April: 4,
+  May: 5,
+  June: 6,
+  July: 7,
+  August: 8,
+  September: 9,
+  October: 10,
+  November: 11,
+  December: 12,
+};
+
+const getMonthScore = (experiences) => {
+  experiences.map((exp) => {
+    exp.start_month_score = monthScore[exp.start_month];
+
+    // exclude end_month === 'none'
+    if (exp.is_current === false) {
+      exp.end_month_score = monthScore[exp.end_month];
+    }
+  });
+
+  return experiences;
+};
+
+const sortExperiences = (experiences) => {
+  const experiencesWithScores = getMonthScore(experiences);
+
+  const sortedExperiences = experiencesWithScores.sort(function(exp1, exp2) {
+    if (exp1.is_current !== exp2.is_current) {
+      return exp1.is_current === true ? -1 : 1;
+    } else {
+      return (
+        exp2.end_year - exp1.end_year ||
+        exp2.end_month - exp1.end_month ||
+        exp2.start_year - exp1.start_year ||
+        exp2.start_month - exp1.start_month
+      );
```

Comment:
Code is much more readable now. Also I'd never considered using the `||` in a sorting function like this, that's very clever (and clean). Good job!

One last thing is even though the code is fairly readable now, it's probably still complicated enough that its worth adding a comment to explain what the sort order is supposed to be (e.g. sort by is_current, then end_date, then start_date) This will save other people (or your future self) time in having to parse through what the function is doing in the future.

I generally recommend doing this with any piece of code that was tricky to write, assuming that if it was tricky to write, it will also probably be tricky to read.